mav_trajectory_generation/src/rpoly.cpp

```diff
@@ -40,30 +40,66 @@
 
 namespace mav_trajectory_generation {
 
-Eigen::VectorXcd findRootsJenkinsTraub(
-    const Eigen::VectorXd& coefficients_increasing) {
-  const Eigen::VectorXd coefficients_decreasing =
-      coefficients_increasing.reverse();
+int findLastNonZeroCoeff(const Eigen::VectorXd& coefficients) {
+  int last_non_zero_coefficient = -1;
+
+  // Find last non-zero coefficient:
+  for (size_t i = coefficients.size() - 1; i != -1; i--) {
+    if (coefficients(i) != 0.0) {
+      last_non_zero_coefficient = i;
+      break;
+    }
+  }
+  return last_non_zero_coefficient;
+}
 
-  const int n_coefficients = coefficients_increasing.size();
+bool findRootsJenkinsTraub(const Eigen::VectorXd& coefficients_increasing,
+                           Eigen::VectorXcd* roots) {
+  // Remove trailing zeros and reverse coefficients.
+  const int last_non_zero_coefficient =
+      findLastNonZeroCoeff(coefficients_increasing);
+  if (last_non_zero_coefficient == -1) {
+    // The polynomial has all zero coefficients and has no roots.
+    roots->resize(0);
+    return true;
+  }
+
+  Eigen::VectorXd coefficients_decreasing =
+      coefficients_increasing.head(last_non_zero_coefficient + 1).reverse();
+
+  const int n_coefficients = coefficients_decreasing.size();
+  if (n_coefficients < 2) {
+    // The polynomial is 0th order and has no roots.
+    roots->resize(0);
+    return true;
+  }
   double* roots_real = new double[n_coefficients];
   double* roots_imag = new double[n_coefficients];
 
   int ret =
       findRootsJenkinsTraub(coefficients_decreasing.data(), n_coefficients - 1,
                             roots_real, roots_imag, NULL);
-
-  Eigen::VectorXcd roots;
-
-  if (ret > 0) {
-    roots.resize(ret);
+  if (ret != -1) {
```

Comment:
Prefer `> -1` as then we don't resize to -2 if they decide on a new return code ;)